src/hamster/lib/datetime.py

```diff
@@ -590,25 +590,33 @@ def from_start_end(cls, start, end=None):
         Convenient to ease backward compatibility,
         and to handle either hdays or datetimes.
         """
+
         if isinstance(start, Range):
             assert end is None, "range and end are mutually exclusive"
             range = start
-        else:
-            if isinstance(start, hday):
-                day = start
-                start = day.start
-                if end is None:
-                    end = day.end
-            elif isinstance(start, pdt.date):
-                # transition from legacy
-                start = hday.from_pdt(start).start
-
-            if isinstance(end, hday):
-                end = end.end
-            elif isinstance(end, pdt.date):
-                end = hday.from_pdt(end).end
-
-            range = Range(start, end)
+        elif isinstance(start, datetime):
+            # this one must come first,
+            # because isinstance(datetime(), date) is True
```

Comment:
I was a bit confused by this comment, since it seems there is no `isinstance(datetime(), date)` below at all. Looking more closely, I see that `pdt` is actually the `datetime` import. Maybe this comment could be adapted too:

    # because isinstance(datetime(), pdt.date) is True

That would have made things more clear for me.

Reply:
`date` inherits from `pdt.date`.
I'll add a clarification though.